[Li Jianhan] iP

jianhandev opened the pull request.

Commit:
Add Gradle support

Commit:
Add Gradle support

jianhandev committed:
Greet, Echo, Exit

Chatbot greets the user, receives and displays the user input. Displays the user a farewell message when 'bye' is entered.

jianhandev committed:
Add, List

Add new tasks to a list. List out all the tasks in the order they were added.

jianhandev committed:
Handle empty strings

Check for empty strings and prevent adding to the list of tasks.

jianhandev committed:
Add Task class

Objects of the Task class stores a description of the task and whether it is completed.

jianhandev committed:
Add ability for task to be marked as done

jianhandev committed:
Add ability for task to be marked as done.

jianhandev committed:
ToDos, Events, Deadlines

Add different types of tasks, namely todo tasks, events and dealines.

jianhandev committed:
ToDos, Events, Deadlines

Add different types of tasks, namely todo tasks, events and dealines.

jianhandev committed:
Automated Text UI Testing

Compares expected and actual output to determine correctness of program.

jianhandev committed:
Automated Text UI Testing

Compares expected and actual output to determine correctness of program.

jianhandev committed:
Handle Errors

Add error handling for invalid inputs and update test files.

jianhandev committed:
Handle Errors

Add error handling for invalid inputs and update test files. Update display formats.

jianhandev committed:
Handle Errors

Add error handling for invalid inputs and update test files. Update display formats.

jianhandev committed:
Merge branch 'master' of https://github.com/jianhandev/ip

jianhandev committed:
Add support for deleting tasks from task list.

jianhandev committed:
Add support for deleting tasks from task list

jianhandev committed:
Add support for deleting tasks from task list

jianhandev committed:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606'

jianhandev committed:
Use enums for user commands

jianhandev committed:
Resolve conflicts

jianhandev committed:
Save the tasks to the hard disk whenever the task list changes

jianhandev committed:
Add dates and times

Replace dates originally represented by strings with java's LocalDate objects.

jianhandev committed:
Save the tasks to the hard disk whenever the task list changes

jianhandev committed:
Integrate new features and resolve conflicts

Add ability to save data to hard disk and add dates and times to master branch.

jianhandev committed:
Handle additional exceptions and update text-ui test files

Handle exceptions raised by out-of-bound index when mark task as done.

jianhandev committed:
Refactor UI components

jianhandev committed:
Refactor and modify TaskList, Storage

jianhandev committed:
Refactor code, extract out closely related code as classes

New classes added: Abstract class Command (and its subclasses), Parser

jianhandev committed:
Add JUnit tests

jianhandev committed:
Package the App as a JAR file

jianhandev committed:
Add JavaDoc comments

jianhandev committed:
Tweak code to comply with coding standard

jianhandev committed:
Add search capability

Give users a way to find a task by searching for a keyword.

jianhandev committed:
Organize the classes into suitable java packages

jianhandev committed:
Fix Level-6 not belonging to any branch

jianhandev committed:
Use Gradle to automate build tasks

jianhandev committed:
Add CheckStyle

YangJiyu98 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,45 @@
+import commands.Command;
+import data.TaskList;
+import exception.ChatbotException;
+import parser.Parser;
+import storage.Storage;
+import ui.Ui;
```
Perhaps you could leave a new line to group your imports?

jianhandev committed:
Add GUI with JavaFX

jianhandev committed:
Fix styles

ianyong commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,23 @@
+    abstract public boolean isExit();
+
+    /**
+     * Executes the command.
+     * @param taskList the TaskList object for handling tasks
+     * @param ui the UI object for text-ui display
+     * @param storage the Storage object for persistence
+     * @throws ChatbotException if execution unsuccessful
```
Should an empty line be left between the description and parameter sections?

ianyong commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,103 @@
+
+                Task task = null;
+
+                switch (type) {
+                    case "T":
+                        task = new Todo(description, isDone);
+                        break;
+                    case "D":
+                        task = new Deadline(description, isDone, LocalDate.parse(timestamp));
+                        break;
+                    case "E":
+                        task = new Event(description, isDone, LocalDate.parse(timestamp));
+                        break;
+                    default:
+                        break;
+                }
```
I believe that the coding standard states that `case` clauses should not have indentation.

ianyong commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,79 @@
+/**
+ * A class representing the command line level user interface. Responsible for reading commands
+ * from the user and displaying data, success/error messages to the user.
+ */
```
Should this documentation block be moved down to right before the class?

ianyong commented on src/main/java/data/Event.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public static Event newEvent(String raw) throws ChatbotException {
+        if (raw.length() == 0) {
+            throw new ChatbotException("Oh boy your event is empty.");
+        }
```
I like your use of a guard clause here! It makes the happy path much more clear.

ianyong reviewed (commented):
Looks good! Quite clean code overall.

jianhandev committed:
Restructure project directory

jianhandev committed:
Add more tests

jianhandev committed:
Add Java Assertions

jianhandev committed:
Chatbot.java: Refactor to adhere to single level of abstraction per function

jianhandev committed:
Refactor repeated code

jianhandev committed:
Refactor code in Parser, Command and Task subclasses

Move code for exception handling from Todo, Event, Deadline classes into Parser class.

jianhandev committed:
Refactor find task by date and keyword using predicates

jianhandev committed:
Improve code quality

jianhandev committed:
Merge pull request #1 from jianhandev/branch-A-Assertions

Add Java Assertions

jianhandev committed:
Merge pull request #2 from jianhandev/branch-A-CodeQuality

Improve code quality

jianhandev committed:
Add Continuous Integration

Create main.yml

jianhandev committed:
Add ability to sort tasks by date

Tasks can be sorted by earliest or latest date using the sort /by command.

jianhandev committed:
Update README.md

jianhandev committed:
Refactor code for parsing commands

Create new interface Parser, to be implemented by parsers for respective commands

jianhandev committed:
Improve exception handling

New ChatbotException subclasses added so that more specific exceptions can be thrown.

jianhandev committed:
Fix styles

jianhandev committed:
main.yml: Remove IO redirection test

jianhandev committed:
Update README.md

jianhandev committed:
Update docs for product website

jianhandev committed:
Polish UI

Change to round profile image, modify font and background color.

jianhandev committed:
Complete User Guide

Add descriptions and examples to each feature.

jianhandev committed:
Add product screenshot

jianhandev committed:
Add more JUnit tests

jianhandev committed:
Fix styles in test files

jianhandev committed:
Move chatbot.txt into new directory ./data

jianhandev committed:
Fix bug that prevents tasks from being saved

jianhandev committed:
Update README.md
